shopfloor_ui_tests/cypress/support/commands.js

```diff
@@ -0,0 +1,300 @@
+                    .type(data.password)
+                    .should("have.value", data.password);
+            } else {
+                cy.get('input[name="password"]').should("have.value", "");
+            }
+        } else {
+            if (data.apikey) {
+                cy.get('input[name="apikey"]')
+                    .type(data.apikey)
+                    .should("have.value", data.apikey);
+            } else {
+                cy.get('input[name="apikey"]').should("have.value", "");
+            }
+        }
+        cy.get('button[type="submit"]').click();
+    });
+});
+
+Cypress.Commands.add("intercept_user_config", () => {
+    cy.intercept({
+        method: "POST",
+        url: "*/app/user_config",
+    }).as("user_config");
+});
+
+Cypress.Commands.add(
+    "wait_for_user_config",
+    ({expect_success = false, reload = false}) => {
+        if (reload) {
+            cy.reload();
+        }
+
+        cy.wait("@user_config").then((res) => {
+            if (expect_success && res.response.statusCode !== 200) {
+                throw new Error(
+                    `Request 'user_config' returned a status of ${res.response.statusCode}. It should have returned a status of 200`
+                );
+            }
+            if (!expect_success && res.response.statusCode < 400) {
+                throw new Error(
+                    `Request 'user_config' returned a status of ${res.response.statusCode}. It should have returned an error status`
+                );
+            }
+            return res;
+        });
+    }
+);
+
+Cypress.Commands.add("compare_sessionStorage_authentication", () => {
+    cy.window().then((win) => {
+        const authenticated = JSON.parse(
+            win.sessionStorage.getItem("shopfloor_authenticated")
+        ).value;
+        if (!authenticated) {
+            throw new Error(
+                "The user isn't authenticated in session storage (shopfloor_authenticated)"
+            );
+        }
+
+        cy.readFile("cypress/fixtures/app_config.json").then((test_appconfig) => {
+            const appconfig = JSON.parse(
+                win.sessionStorage.getItem("shopfloor_appconfig")
+            ).value;
+            if (JSON.stringify(appconfig) !== JSON.stringify(test_appconfig)) {
+                throw new Error(
+                    "The user's information doesn't match the information in the response received from the login request"
+                );
+            }
+        });
+    });
+});
+
+// Commands for the navigation tests
+
+Cypress.Commands.add("sidebar_menu_to", (toClick = "") => {
+    cy.get(".v-toolbar__content").children("button").click();
+    cy.get('a[href="#/"]');
+    cy.get('a[href="#/settings"]');
+    cy.get('a[href="#/scan_anything"]');
+    cy.get(`a[href="#/${toClick}"]`).click();
+    cy.get("header").click({force: true});
+});
+
+// Commands for the profile tests
+
+Cypress.Commands.add("check_profile_list", () => {
```

Comment:
this command should accept the list of profiles you expect and not having them hardcoded